aries_cloudagent/connections/models/diddoc/util.py

```diff
@@ -74,27 +50,13 @@ def canon_ref(did: str, ref: str, delimiter: str = None):
             introducing identifier (';') against DID resource
     """
 
-    if not ok_did(did):
-        raise ValueError("Bad DID {} cannot act as DID document identifier".format(did))
-
-    if ok_did(ref):  # e.g., LjgpST2rjsoxYegQDRm7EL
-        return "did:sov:{}".format(did)
-
-    if ok_did(resource(ref, delimiter)):  # e.g., LjgpST2rjsoxYegQDRm7EL#keys-1
-        return "did:sov:{}".format(ref)
-
-    if ref.startswith(
-        "did:sov:"
-    ):  # e.g., did:sov:LjgpST2rjsoxYegQDRm7EL, did:sov:LjgpST2rjsoxYegQDRm7EL#3
-        rv = ref[8:]
-        if ok_did(resource(rv, delimiter)):
-            return ref
-        raise ValueError("Bad URI {} does not correspond to a sovrin DID".format(ref))
-
-    if urlparse(ref).scheme:  # e.g., https://example.com/messages/8377464
+    if (
+        ref.startswith("did:")
+        or urlparse(ref).scheme  # e.g., https://example.com/messages/8377464
+    ):
         return ref
 
-    return "did:sov:{}{}{}".format(did, delimiter if delimiter else "#", ref)  # e.g., 3
+    return "{}{}{}".format(did, delimiter if delimiter else "#", ref)
```

Comment:
return f"{did}{delimiter or '#'}{ref}" is slightly briefer but it is not so important